### DISCUSSION TITLE
handling irregular sampling locations

### DISCUSSION THREAD

**Discussion by rgcoe** [pydata/xarray#6503] [Q&A]
I have a geospatial dataset in which the data is reported at irregular (not sure if this is the right word?) lat/lon points. I'm relatively new to geospatial analyses, but I've found `xarray` very helpful in the past when working with gridded datasets. Can anyone point me to some guidance on how to best handle my dataset with irregular sampling points. One thing I'd like to do is take a spatial average (e.g., `ds.mean(['lat','lon']`). The following tutorials appear to give some methods and packages (`datashader`, `geopandas`) for accomplishing this, but I'm finding this a bit clunky and thinking that maybe there's a better way?

- https://james-brennan.github.io/posts/fast_gridding_geopandas/
- https://geographicdata.science/book/notebooks/03_spatial_data.html#tables-as-surfaces

Here's a sort of MWE for the dataset I'm dealing with:

```python
n = 10
da = xr.DataArray(np.random.rand(n),
                  coords={
                      'lat':('location',np.arange(n)*np.random.rand(n)),
                      'lon':('location',np.arange(n)*np.random.rand(n)+150),
                      },
                  dims='location')
```

![image](https://user-images.githubusercontent.com/5589887/164284682-f51625c5-28a9-45c1-8f66-09f3ef3c2976.png)

**Comment by keewis** [comment]
I guess your algorithm is a bit more complicated than `ds.mean(dim=["lat", "lon"])`, but with the example you posted I think it should be fine to use `ds.mean(dim="location")` instead.

You might also want to look into [`verde`](https://www.fatiando.org/verde/latest/) which provides some more advanced features for working with this kind of irregular data.

**Reply by benbovy** [reply]
If nearest-neighbor interpolation is enough for you you can also look into [xoak](http://xoak.readthedocs.io).

**Reply by rgcoe** [reply]
This is helpful, thanks! I did find `verde` and it does what I was looking for. The trick for me now is that my data also has a time dimension, so I basically want to apply the transformation from the irregular points to regular grid for `n` time samples. This seems like the sort of thing I should be using [`apply_ufunc`](https://xarray.pydata.org/en/stable/generated/xarray.apply_ufunc.html) or [`map_blocks`](https://xarray.pydata.org/en/stable/generated/xarray.map_blocks.html#xarray.map_blocks) for, but I'm struggling to get this to run as fast as it should (with `dask`). Thanks again for your help!
